### PR TITLE
[DOC] Un-deprecate join_apply (#1399)

### LINKED ISSUE #1399: Proposed alternative for `join_apply` in deprecation notice does not replicate its behavior
# Brief Description

Currently, the docs for `join_apply` state its deprecation, and advise to use `transform_column` instead. However, `join_apply` works row-wise, and `transform_column` is either single-column or multiple columns but with the same function applied to each column individually. 

At this point, it is unclear what would be the correct replacement for `join_apply`. A reference to an alternative approach, or a snippet would make the documentation clearer. 

For the record, I filed this under "Documentation fix" as it's not a code problem, but a documentation problem.

A possible alternative may be to suggest the actual code `join_apply` had:

```
df = df.copy().join(df.apply(fn, axis=1).rename(new_column_name))
```

# Relevant Context

- [Link to documentation page](https://pyjanitor-devs.github.io/pyjanitor/api/functions/#janitor.functions.join_apply)

### PR DESCRIPTION
**Pull request by lbeltrame**
# PR Description

There is no current alternative to row-wise operations, so revert the deprecation for now as discussed in issue #1399.

**This PR resolves #1399 .**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.
2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.
3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl
- @samukweku

### CODE REVIEW

**Review by samukweku — APPROVED** [review]

**Comment by samukweku** [comment]
@lbeltrame looks good. Just needs one more review and we will merge if all goes well.

**Review by ericmjl — APPROVED** [review]

**Comment by ericmjl** [comment]
@lbeltrame thank you for the contribution! @samukweku would you be kind enough to cut a new release please?

**Comment by samukweku** [comment]
@ericmjl  ok to make it as a patch? or minor?

**Pull request merged**
